amida-tech/indaba-client
# Feature/inba 852 jf react router update

#### What does this PR do?
This PR updates react-router to version 4.0, bringing us up to the latest (or very near) react-router handling. 

#### Related JIRA tickets:
Will be useful for future implementation of INBA-804 and INBA-782. 
Also note the error associated with INBA-854. This existed before this update.

#### How should this be manually tested?
You will need to run `yarn` to install several new packages. After that, this is a totally regression test of pretty much everything to ensure that nothing breaks. Please check messaging with particular care in case I screwed up how the containers are handled. And pay particular attention to the console log for links and anything that "pushes" a new URL (ie transition from login to projects or tasks). 

The "Private Route" is something I'm planning for INBA-804. Sorry I left it in there, but I will either use it or delete it shortly. 

#### Background/Context

#### Screenshots (if appropriate):
